### PR TITLE
ENH: add convinence node smap, for splaying out tuples into map

### PR DESCRIPTION
**Pull request by CJ-Wright**
Attn @jrmlhermitte

### CODE REVIEW

**Comment by mrocklin** [comment]
I think that the conventional name for this in Python is `starmap`.  See `itertools`.  Also, I suspect that your current solution is Python 2 only.

Otherwise this looks good to me.

File: streamz/core.py [hunk 1]
```diff
@@ -490,6 +490,50 @@ def update(self, x, who=None):
...
+        stream_name = kwargs.pop('stream_name', None)
+        self.kwargs = kwargs
+        self.args = args
+
+        Stream.__init__(self, upstream, stream_name=stream_name)
+
+    def update(self, x, who=None):
+        y = x + self.args
+        try:
+            result = self.func(*y, **self.kwargs)
```
[the remark this answers is not in the document]
Comment (CJ-Wright) [reply streamz/core.py]: Only in Python 3.

**Comment by mrocklin** [comment]
This looks good to me.  You should probably add an entry to `docs/source/api.rst`

**Comment by CJ-Wright** [comment]
@jrmlhermitte I'll put dstarmap in a separate PR

**Comment by mrocklin** [comment]
This looks great to me.  Merging.  Thanks @CJ-Wright !

**Comment by CJ-Wright** [comment]
Thank you @mrocklin!

**Comment by jrmlhermitte** [comment]
This is great very useful thanks @CJ-Wright!

**Comment by CJ-Wright** [comment]
@jrmlhermitte it was your suggestion!

**Pull request merged**